builder/builder.go

```diff
@@ -572,36 +596,112 @@ func (b *Builder) BuildUpdate(prefixflag string, minVersion int, format string,
 		fmt.Println("Signed Manifest.MoM")
 	}
 
+	fullChrootDir := filepath.Join(b.Statedir, "image", b.Mixver, "full")
+	if _, err = os.Stat(fullChrootDir); err != nil {
+		return errors.Wrapf(err, "couldn't access the full chroot")
+	}
+
+	outputDir := filepath.Join(b.Statedir, "www")
+	if _, err = os.Stat(outputDir); err != nil {
+		return errors.Wrapf(err, "couldn't access the swupd output dir (www)")
+	}
+
 	// Create full files.
-	fullfilecmd := exec.Command(prefixflag+"swupd_make_fullfiles", "-S", b.Statedir, b.Mixver)
-	fullfilecmd.Stdout = os.Stdout
-	fullfilecmd.Stderr = os.Stderr
-	err = fullfilecmd.Run()
-	if err != nil {
-		return errors.Wrapf(err, "couldn't create fullfiles")
+	timer.Start("create fullfiles")
+	if UseNewSwupdServer {
+		manifestFile := filepath.Join(b.Statedir, "www", b.Mixver, "Manifest.full")
+		if _, err = os.Stat(manifestFile); err != nil {
+			return errors.Wrapf(err, "couldn't access the full manifest")
+		}
+		var m *swupd.Manifest
+		m, err = swupd.ParseManifestFile(manifestFile)
+		if err != nil {
+			return errors.Wrapf(err, "couldn't read full manifest")
+		}
+		fullfilesDir := filepath.Join(outputDir, b.Mixver, "files")
+		err = os.MkdirAll(fullfilesDir, 0777)
+		if err != nil {
+			return errors.Wrapf(err, "couldn't create the full files directory")
+		}
+		err = swupd.CreateFullfiles(m, fullChrootDir, fullfilesDir)
+		if err != nil {
+			return err
+		}
+	} else {
+		fullfilecmd := exec.Command(prefixflag+"swupd_make_fullfiles", "-S", b.Statedir, b.Mixver)
+		fullfilecmd.Stdout = os.Stdout
+		fullfilecmd.Stderr = os.Stderr
+		err = fullfilecmd.Run()
+		if err != nil {
+			return errors.Wrapf(err, "couldn't create fullfiles")
+		}
 	}
+	timer.Stop()
 
 	// Create zero packs.
-	zeropackArgs := []string{"--to", b.Mixver, "-S", b.Statedir}
-	if prefixflag != "" {
-		zeropackArgs = append(zeropackArgs, "--repodir", prefixflag)
-	}
-	zeropackcmd := exec.Command("mixer-pack-maker.sh", zeropackArgs...)
-	zeropackcmd.Stdout = os.Stdout
-	zeropackcmd.Stderr = os.Stderr
-	err = zeropackcmd.Run()
-	if err != nil {
-		return errors.Wrapf(err, "couldn't create zero packs")
+	timer.Start("create zero packs")
+	if UseNewSwupdServer {
+		var bundles map[string]*swupd.BundleToPack
+		bundles, err = swupd.FindBundlesToPack(0, mixVerUint, b.Statedir)
+		if err != nil {
+			return errors.Wrapf(err, "couldn't find bundles to pack")
+		}
+
+		// TODO: Use goroutines.
+		for _, b := range bundles {
```

Comment:
Are you shadowing the `b *Builder` receiver on purpose?

Reply:
Not really. Will fix this.